apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs

```diff
@@ -46,6 +116,159 @@ private SpannerConversionOptions(bool useDBNull)
+            var clone = Clone();
+            clone.UseDBNull = !useClrDefaultToNull;
+            return clone;
+        }
+
+        internal SpannerConversionOptions WithClrToSpannerMappings(string clrToSpannerMappings)
+        {
+            var clone = Clone();
+            ValidateAndParseClrToSpannerTypeMappings(clrToSpannerMappings, clone);
+            return clone;
+        }
+
+        internal SpannerConversionOptions WithSpannerToClrMappings(string spannerToClrMappings)
+        {
+            var clone = Clone();
+            ValidateAndParseSpannerToClrTypeMappings(spannerToClrMappings, clone);
+            return clone;
+        }
+
+        private void ValidateAndParseSpannerToClrTypeMappings(string input, SpannerConversionOptions options)
+        {
+            // Empty string or null is valid to unset the values.
+            if (string.IsNullOrEmpty(input))
+            {
+                // Set defaults.
+                options.SetSpannerToClrTypeDefaults();
+                return;
+            }
+
+            // Get all type mappings.
+            var mappings = input.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+
+            // Validate that mappings are valid.
+            if (mappings == null || !mappings.Any())
+            {
+                throw new ArgumentException($"'{input}' is not a valid value for ${nameof(SpannerConnectionStringBuilder.SpannerToClrTypeDefaultMappings)}");
+            }
+
+            // Validate that mapping matches one of the pre-defined values.
+            var invalidMappings = mappings.Except(AllowedSpannerToClrTypeMappings, StringComparer.OrdinalIgnoreCase);
+            if (invalidMappings.Any())
+            {
+                throw new ArgumentException($"The following mappings are invalid:'{string.Join(",", invalidMappings)}' for ${nameof(SpannerConnectionStringBuilder.SpannerToClrTypeDefaultMappings)}");
+            }
+
+            // Check multiples for each type.
+            // Currently, we have Float64 and Date to check.
+            if (mappings.Count(j => j.StartsWith("Float64", StringComparison.OrdinalIgnoreCase)) > 1)
+            {
+                throw new ArgumentException($"'{input}' is not a valid value as multiple mappings from Float64 to CLR type are provided for ${nameof(SpannerConnectionStringBuilder.SpannerToClrTypeDefaultMappings)}");
+            }
+
+            if (mappings.Count(j => j.StartsWith("Date", StringComparison.OrdinalIgnoreCase)) > 1)
+            {
+                throw new ArgumentException($"'{input}' is not a valid value as multiple mappings from Date to CLR type are provided for ${nameof(SpannerConnectionStringBuilder.SpannerToClrTypeDefaultMappings)}");
+            }
+
+            // If we reach here, all is well.
+            foreach (var mapping in mappings)
+            {
+                // Discard pattern, primarily to simplify code using switch expressions.
+                _ = mapping switch
```

Comment:
This will be case-sensitive, having done a case-insensitive comparison earlier.
We could either make the comparisons case-sensitive, or lower-case the input and expected values, or use if statements.